### DISCUSSION TITLE
Need assistance for RAG with Business process workflow in Semantic kernel

### DISCUSSION THREAD

**Discussion by rajeshbayya** [microsoft/semantic-kernel#9888] [General]
Currently we are having RAG solution using semantic kernel in c# which searches in user guide document and gives answer to user. I want to a business process workflow to existing semantic kernel solution. Business workflow is run the tax process and wait till it completes. Once process it is completed it should print tax forms pdf automatically in chat. I have rest apis for the starting tax process, process status and print tax forms.
Please assist "How to achieve this along with the existing RAG solution".
RAG is implemented using the refrence https://github.com/Azure-Samples/azure-search-openai-demo-csharp @dmytrostruk 
@juliomenendez @atifaziz @dandv @

**Comment by moonbox3** [comment]
Tagging @alliscode for visibility with C# Processes.

**Comment by sphenry** [comment]
@rajeshbayya Take a look at the Process Framework here: [Process Framework Docs](https://learn.microsoft.com/en-us/semantic-kernel/frameworks/process/process-framework) and the Samples here: [Process Framework Samples](https://github.com/microsoft/semantic-kernel/tree/main/dotnet/samples/GettingStartedWithProcesses)

**Comment by AftonstraleIT** [comment]
I was attending a presentation in early November in Malmoe, Sweden; held by Matthew Bolanos. He was presenting several things, including the use of agents/assistants. During his presentation he also introduced the ProcessBuilder, which is also used in your samples linked in the reply from @sphenry.

I am using latest (version 1.32.0) of Microsoft.SemanticKernel and nowhere is the namespace for .Process to be found (thus I cannot use the ProcessBuilder). This makes the samples impossible to run.

Do I need to do something special because this is an experimental feature?
Is there any kind of name changes not yet reflected in the sample code?
Or maybe something completely different?

Please advice.

**Edit:**
For clarification, I have tried activating experimental functions by adding the below to my .csproj file and I can see that it is accepting this, but still no ProcessBuilder anywhere.

```
<DefineConstants>EXPERIMENTAL;SKEXP0080</DefineConstants>
<NoWarn>
$(NoWarn);CS8618,IDE0009,CA1051,CA1050,CA1707,CA1054,CA2007,VSTHRD111,CS1591,RCS1110,RCS1243,CA5394,SKEXP0001,SKEXP0010,SKEXP0020,SKEXP0040,SKEXP0050,SKEXP0060,SKEXP0070,SKEXP0080,SKEXP0101,SKEXP0110,OPENAI001
</NoWarn>
```

**Reply by AftonstraleIT** [reply]
**SOLVED**

I have tried quite a lot of things here, and once I added a few extra nuget packages it was working fine. Since I did a lot of changes, I am not sure if this is the single point of failure, but at least I have detected a flaw in the documentation.

In this page, I was expecting a list of dependent nuget packages, similar as listed for the Agent Framework just above. But since there were none listed I was expecting it to be in the base package.
https://learn.microsoft.com/en-us/semantic-kernel/frameworks/process/process-framework?utm_source=chatgpt.com

But when searching all over the place, I found the missing pieces in the github read-me:
https://github.com/microsoft/semantic-kernel/tree/main/dotnet/samples/GettingStartedWithProcesses

Here it states that I need
- Microsoft.SemanticKernel.Process.Abstractions
- Microsoft.SemanticKernel.Process.Core
- Microsoft.SemanticKernel.Process.LocalRuntime

So by activating download of alpha releases and retrieving these packages I can now at least compile my project.

Please update documentation to allow other users to save many hours of frustration.

By the way, thank you for a nice packager. :-)
